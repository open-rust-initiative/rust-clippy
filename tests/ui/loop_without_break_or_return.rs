#![allow(unused, clippy::never_loop)]
#![warn(clippy::loop_without_break_or_return)]

fn test_01() {
    loop {
        println!("Hello, Rust!");
    }

    loop {
        break;
    }

    'outer: loop {
        break 'outer;
    }

    'outer: loop {
        break;
    }
}

fn test_02() {
    loop {
        if 2 < 3 {
            break;
        }
    }
}

fn test_03() {
    'outer1: loop {
        for x in 0..5 {
            if x == 3 {
                break 'outer1;
            }
        }
    }

    'outer2: loop {
        for x in 0..5 {
            if x == 3 {
                break;
            }
        }
    }

    'outer3: loop {
        for x in 0..5 {
            if x == 3 {
                println!("Hello, Rust!");
            } else {
                break 'outer3;
            }
        }
    }
}

fn test_04() {
    'outer1: loop {
        loop {
            println!("Hello, Rust!");
        }
        break;
    }

    'outer2: loop {
        loop {
            break;
        }
    }

    'outer3: loop {
        loop {
            break 'outer3;
        }
    }

    'outer4: loop {
        'inner: loop {
            loop {
                break 'inner;
            }
        }
    }

    'outer5: loop {
        loop {
            'inner: loop {
                loop {
                    loop {
                        break 'inner;
                    }
                    break 'outer5;
                }
            }
        }
    }
}

fn main() {
    // test code goes here
}
