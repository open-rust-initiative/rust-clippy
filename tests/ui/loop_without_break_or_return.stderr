error: consider adding `break` or `return` statement in the loop block
  --> $DIR/loop_without_break_or_return.rs:5:5
   |
LL | /     loop {
LL | |         println!("Hello, Rust!");
LL | |     }
   | |_____^
   |
   = note: `-D clippy::loop-without-break-or-return` implied by `-D warnings`

error: consider adding `break` or `return` statement in the loop block
  --> $DIR/loop_without_break_or_return.rs:39:5
   |
LL | /     'outer2: loop {
LL | |         for x in 0..5 {
LL | |             if x == 3 {
LL | |                 break;
LL | |             }
LL | |         }
LL | |     }
   | |_____^

error: consider adding `break` or `return` statement in the loop block
  --> $DIR/loop_without_break_or_return.rs:60:9
   |
LL | /         loop {
LL | |             println!("Hello, Rust!");
LL | |         }
   | |_________^

error: consider adding `break` or `return` statement in the loop block
  --> $DIR/loop_without_break_or_return.rs:66:5
   |
LL | /     'outer2: loop {
LL | |         loop {
LL | |             break;
LL | |         }
LL | |     }
   | |_____^

error: consider adding `break` or `return` statement in the loop block
  --> $DIR/loop_without_break_or_return.rs:78:5
   |
LL | /     'outer4: loop {
LL | |         'inner: loop {
LL | |             loop {
LL | |                 break 'inner;
LL | |             }
LL | |         }
LL | |     }
   | |_____^

error: consider adding `break` or `return` statement in the loop block
  --> $DIR/loop_without_break_or_return.rs:87:9
   |
LL | /         loop {
LL | |             'inner: loop {
LL | |                 loop {
LL | |                     loop {
...  |
LL | |             }
LL | |         }
   | |_________^

error: aborting due to 6 previous errors

